src/test/java/uk/gov/hmcts/reform/em/orchestrator/stitching/StitchingServiceTest.java

```diff
@@ -41,6 +41,24 @@ public void stitchSuccessful() throws StitchingServiceException, InterruptedExce
         CcdDocument docId = service.stitch(new CcdBundleDTO(), "token");
 
         Assert.assertEquals(docId.getUrl(), "AAAAAA");
+        Assert.assertEquals(docId.getFileName(), "stitched.pdf");
+    }
+
+    @Test
+    public void stitchSuccessfulWithFileName() throws StitchingServiceException, InterruptedException {
+        List<String> responses = new ArrayList<>();
+        responses.add("{ id: 1, taskState: 'NEW', bundle: { stitchedDocumentURI: null } }");
+        responses.add("{ id: 1, taskState: 'NEW', bundle: { stitchedDocumentURI: null } }");
+        responses.add("{ id: 1, taskState: 'NEW', bundle: { stitchedDocumentURI: null } }");
+        responses.add("{ id: 1, taskState: 'NEW', bundle: { stitchedDocumentURI: null } }");
+        responses.add("{ id: 1, taskState: 'DONE', bundle: { stitchedDocumentURI: 'AAAAAA', fileName: 'a.pdf' } }");
```

Comment:
why are there 5 lines here? should it only look at the latest one? if so, why isn't that in a separate test?

Reply:
replicates the polling that happens with the stitching API